model/host/host.go

```diff
@@ -2461,3 +2461,167 @@ func (h *Host) IsSubjectToHostCreationThrottle() bool {
+		{
+			"$match": bson.M{StatusKey: bson.M{"$ne": evergreen.HostTerminated}},
+		},
+		{
+			"$lookup": bson.M{
+				"from":         task.Collection,
+				"localField":   RunningTaskKey,
+				"foreignField": task.IdKey,
+				"as":           "task_full",
+			},
+		},
+		{
+			"$unwind": bson.M{
+				"path":                       "$task_full",
+				"preserveNullAndEmptyArrays": true,
+			},
+		},
+	}
+
+	// CONTEXT
+	env := evergreen.GetEnvironment()
+	ctx, cancel := env.Context()
+	defer cancel()
+
+	// TOTAL RUNNING HOSTS COUNT
+	countPipeline := []bson.M{}
+	for _, stage := range runningHostsPipeline {
+		countPipeline = append(countPipeline, stage)
+	}
+	countPipeline = append(countPipeline, bson.M{"$count": "count"})
+
+	tmp := []counter{}
+	cursor, err := env.DB().Collection(Collection).Aggregate(ctx, countPipeline)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	err = cursor.All(ctx, &tmp)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	totalRunningHostsCount := 0
+	if len(tmp) > 0 {
+		totalRunningHostsCount = tmp[0].Count
+	}
+
+	// APPLY FILTERS TO PIPELINE
+	hasFilters := false
+
+	if len(hostID) > 0 {
+		hasFilters = true
+
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$match": bson.M{
+				IdKey: hostID,
+			},
+		})
+	}
+
+	if len(distroID) > 0 {
+		hasFilters = true
+
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$match": bson.M{
+				"distro._id": distroID,
+			},
+		})
+	}
+
+	if len(currentTaskID) > 0 {
+		hasFilters = true
+
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$match": bson.M{
+				RunningTaskKey: currentTaskID,
+			},
+		})
+	}
+
+	if len(startedBy) > 0 {
+		hasFilters = true
+
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$match": bson.M{
+				StartedByKey: startedBy,
+			},
+		})
+	}
+
+	if len(statuses) > 0 {
+		hasFilters = true
+
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$match": bson.M{
+				StatusKey: bson.M{"$in": statuses},
+			},
+		})
+	}
+
+	// APPLY SORTERS TO PIPELINE
+	sorters := bson.D{}
+	if len(sortBy) > 0 {
+		sorters = append(sorters, bson.E{Key: sortBy, Value: sortDir})
+	}
+	// _id must be the LAST item in sort array to ensure a consistent sort order when previous sort keys result in a tie
+	sorters = append(sorters, bson.E{Key: IdKey, Value: 1})
+	runningHostsPipeline = append(runningHostsPipeline, bson.M{
+		"$sort": sorters,
+	})
+
+	// APPLY PAGINATION
+	if limit > 0 {
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$skip": page * limit,
+		})
+		runningHostsPipeline = append(runningHostsPipeline, bson.M{
+			"$limit": limit,
+		})
+	}
+
+	hosts := []Host{}
+
+	cursor, err = env.DB().Collection(Collection).Aggregate(ctx, runningHostsPipeline)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	err = cursor.All(ctx, &hosts)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+
+	// FILTERED HOSTS COUNT
+	var filteredHostsCount *int
```

Comment:
it looks like page and limit are applied to the pipeline before getting the filteredHostsCount. I think these orders should be flipped so we can get the filtered host count unrestricted by pagination

Reply:
nice one! updated.